### ISSUE TITLE
[3.3.5] Moss covered feet does not make user fall.

### ISSUE THREAD

**Issue by Smileynator** [TrinityCore/TrinityCore#19022]
[//]: # (***************************************************************)
[//]: # (** DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED **)
[//]: # (***************************************************************)

**Description:**
Elder Lake Creeper (DB GUID 19099) casts a spell with an effect named "Moss Covered Feet".
Description dictates the user can randomly fall over when hit. The spell is there, it covers the player character in green, but the effect never happens.

- http://www.wowhead.com/npc=1956/elder-lake-creeper
- http://wotlk.openwow.com/npc=1956 (3.3.5 info)
- http://www.wowhead.com/spell=6870/moss-covered-feet
- http://wotlk.openwow.com/spell=6870 (3.3.5 info)
- http://www.wowhead.com/spell=6871/moss-covered-feet-proc
- http://wotlk.openwow.com/spell=6871 (3.3.5 info)

Ref: http://puu.sh/tJZL7.png
NPC info: http://puu.sh/tJZSf.png

**Current behaviour:** (Tell us what happens.)
User does not sometimes fall over when hit.

**Expected behaviour:** (Tell us what should happen instead.)
User should sometimes fall over when hit.

**Steps to reproduce the problem:**

1. Summon NPC
2. Let it cast the spell on you
3. Continue to be hurt by NPC
4. See that nothing happens even after 100 hits.

**Branch(es):** 3.3.5

**TC rev. hash/commit:**  39b20b4431a1a67446e9cc4c2221fd42cc907291

**TDB version:**  TDB 335.62 + updates up to and including 2017_01_17_00_world.sql

**Operating system:**  Windows 7

[//]: # (This template is for problem reports. For other types of report, edit it accordingly.)
[//]: # (If this is a crash report, include the crashlog with https://gist.github.com/)
[//]: # (For fixes containing C++, create a Pull Request.)

**Comment by tkrokli** [comment]
Changes made to the post above:
- added links to wowhead & wotlk.openwow pages
- specified branch as 3.3.5 (creature removed in patch 4.0.3a)
- minor layout edits

This effect should (in theory) also be used by the creature Lagoon Walker (http://www.wowhead.com/npc=20291/lagoon-walker) in Zangarmarsh. Could you check if this spell effect works if the spell is cast from that creature too?

**Comment by Smileynator** [comment]
Thanks for the extra detective work, i will check that fact.

**Comment by tkrokli** [comment]
Related spells (instant cast presumably made by the NPC) :
- http://www.wowhead.com/spell=31399/moss-covered-feet
- http://www.wowhead.com/spell=31398/moss-covered-feet-proc

**Comment by offl** [comment]
Fall down(ID 6869) is used as the effect of 6870(Chance to **fall down** when hit).
Another spell is used as the effect of 6866 (Chance to **drop melee weapon** when attacking). But I can't find it. You can read the description about this spell [here](http://www.wowhead.com/npc=3780/singed-shambler#comments:id=492512:reply=77276).

Maybe Fall down(ID 6869) is used also for 31399(Chance to **fall down** when hit). Don't know.
> 31396	Moss Covered Hands Proc
31397	Moss Covered Hands
31398	Moss Covered Feet Proc
31399	Moss Covered Feet

Also currently NPCs casts triggered spells instead of spells which should trigger spell(common bug). 
Should be like that(SAI actions should be removed):
```
 entry  auras           
------  --------------
  1953  6867
  1954  6867
  1955  6871
  1956  6871
  3780  6867 6871
```

**Comment by Smileynator** [comment]
@tkrokli I tested it in zangar http://puu.sh/tMTSV.png And being hit by there guys for a minute or 5, by 5 of them. I have not seen them cast the spell once. So one must assume that they do not know this spell or are somehow unable to use the spell. Wowhead does say these should have that spell though.

Test was done in the same branch etc. as described in original issue post.

**Comment by tkrokli** [comment]
Oh well, maybe it is just another example of spell proc not working, I don't know. @ariel- usually works on things like this in his spare time, given that it is useful and meaningful to most of the players, especially if it is game-breaking (in this case it is not). In the old labeling system, I would have used [Priority-Cosmetic] based on how seldom players actually find this particular case to be important in the game. I can't promise much, unless there is a simple solution to fix this issue.

**Comment by offl** [comment]
> I tested it in zangar

On retail?

**Comment by offl** [comment]
> And being hit by there guys for a minute or 5, by 5 of them. I have not seen them cast the spell once.

Some spells procs only if **you** hit the creature.

**Comment by Smileynator** [comment]
@offl I was in the assumption that this was a spell cast by the creature during battle, not something i could get from hitting it myself. I could try again to make sure then.

On the subject of working on this. I get that it is cosmetic and small, but broken is broken. It does not need priority, but in the end we all want the game to feel like Retail, and these are eventually the things that make it that way. (like cosmetic effects that happen during quests etc. npc's that chat and walk, they are all like this)

**Comment by ariel-** [comment]
Spell is a dummy proc aura, we need to check sniffs to script it's proper behavior

**Comment by ariel-** [comment]
No sniffs of the spell, would be good if someone can sniff this and upload to forum 👍

**Comment by Shauren** [comment]
These are not in the game anymore

**Comment by ariel-** [comment]
What about http://www.wowhead.com/npc=20291/lagoon-walker?

**Comment by Smileynator** [comment]
Those are still there and can be sniffed. If someone has retail and time.

**Comment by Elarose** [comment]
Made a sniff, uploaded, hitting post gives me permission error, hope thats only because of it being hidden for devs. If it didn't actually send let me know.

**Comment by Aokromes** [comment]
don't upload sniffs as attachments, it stops us to use mass downloaders to get sniffs.

**Comment by ariel-** [comment]
@Elarose: thanks for taking your time getting the info!

@offl: Was indeed the correct spell, I'll use it for 31399 too, since no other "Fall down" spell was found, and the creatures using it were removed.

As for "Moss Covered Hand", the proc could be any of the "Dropped Weapon" spells (Chance to drop melee weapon, in this case there are a few similar spells with different durations)

**Comment by tkrokli** [comment]
And thanks to all who have contributed to make this improvement to the source. 👍

**Comment by Elarose** [comment]
@ariel-  If it will fix something in 3.3.5 branch and is still the same on retail, just ask -anytime- if you need a sniff, you got it.
